agreement_rebate/models/agreement_rebate_settlement.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2020 Tecnativa - Carlos Dauden
+# Copyright 2020 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, models, fields
+from odoo.osv import expression
+from odoo.tools import safe_eval
+from odoo.exceptions import UserError
+
+
+class AgreementRebateSettlement(models.Model):
+    _name = 'agreement.rebate.settlement'
+    _description = 'Agreement Rebate Settlement'
+    _order = 'date DESC'
+
+    name = fields.Char()
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        required=True,
+        index=True,
+        default=lambda self: self.env.user.company_id.id,
+    )
+    date = fields.Date(default=fields.Date.today())
+    date_from = fields.Date()
+    date_to = fields.Date()
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Partner',
+    )
+    line_ids = fields.One2many(
+        comodel_name='agreement.rebate.settlement.line',
+        inverse_name='rebate_settlement_id',
+        string='Settlement Lines',
+    )
+    amount_invoiced = fields.Float(string='Amount invoiced')
+    amount_rebate = fields.Float(string='Amount rebate')
```

Comment:
```suggestion
    total_amount = fields.Float(string='Amount rebate')
```
and `digits`??
Why not `fields.Monetary`? Is there conversion between currencies?